fem/transfer.hpp

```diff
@@ -244,6 +245,126 @@ class L2ProjectionGridTransfer : public GridTransfer
    virtual const Operator &BackwardOperator();
 };
 
+
+/** @brief Transfer data in H1 finite element spaces between a coarse mesh and
+    an embedded refined mesh using L2 projection. */
+/** The forward, coarse-to-fine, transfer uses L2 projection. The backward,
+    fine-to-coarse, transfer is defined as B = (F^t M_f F)^{-1} F^t M_f, where
+    F is the forward transfer matrix, and M_f is a diagonal (lumped) mass matrix
+    computed through row-summation. Note that the backward transfer operator, B,
+    is a left inverse of the forward transfer operator, F, i.e. B F = I. Both F
+    and B are defined in physical space.
+
+    This class supports H1 finite element spaces and fine meshes that are a
+    uniform refinement of the coarse mesh.  Generally, the coarse and fine FE
+    spaces can have different orders, however, in order for the backward
+    operator to be well-defined, the number of fine dofs (in a coarse element)
+    should not be smaller than the number of coarse dofs.
+
+    For L2 finite element spaces, see L2ProjectionGridTransfer. */
+class L2ProjectionH1GridTransfer : public GridTransfer
+{
+protected:
+   /** Class representing projection operator between a high-order H1 finite
+       element space on a coarse mesh, and a low-order H1 finite element space
+       on a refined mesh (LOR). We assume that the low-order space, fes_lor,
+       lives on a mesh obtained by refining the mesh of the high-order space,
+       fes_ho. */
+   class L2ProjectionH1 : public Operator
+   {
+      const FiniteElementSpace& fes_ho;
+      const FiniteElementSpace& fes_lor;
+
+      SparseMatrix* R;
+      SparseMatrix M_LH;
+      mutable DenseMatrix* P;
```

Comment:
```suggestion
      SparseMatrix R;
      SparseMatrix M_LH;
      mutable DenseMatrix P;
```

These pointers can be replaced with objects, and when `P` is constructed, just use `SetSize`, and when `R` is allocated, you can use `SparseMatrix::Swap` to set up the proper sizes.

Reply:
Agreed.  Fixed in the latest commit.